test/ITensorAutoHOOT/runtests.jl

```diff
@@ -1,4 +1,205 @@
 using ITensorNetworkAD
-using Test
+using AutoHOOT
+using ITensors
+using Zygote
 
-@testset "ITensorAutoHOOT.jl" begin end
+const go = AutoHOOT.graphops
+const itensorah = ITensorNetworkAD.ITensorAutoHOOT
+
+@testset "test interface" begin
+  i = Index(2, "i")
+  j = Index(3, "j")
+  k = Index(2, "k")
+  l = Index(4, "l")
+
+  A = randomITensor(i, j)
+  B = randomITensor(j, k)
+  C = randomITensor(k, l)
+
+  out = A * B * C
+  network = [A, C, B]
+
+  nodes, dict = itensorah.generate_einsum_expr([network])
+  network = itensorah.extract_network(nodes[1], dict)
+  out2 = network[1] * network[2] * network[3]
+  @test isapprox(storage(out), storage(out2))
+end
+
+@testset "test compute" begin
+  i = Index(2, "i")
+  j = Index(3, "j")
+  k = Index(4, "k")
+  l = Index(5, "l")
+  m = Index(6, "m")
+
+  A = randomITensor(i, j)
+  B = randomITensor(j, k)
+  C = randomITensor(k, l)
+  D = randomITensor(l, m)
+  E = randomITensor(m, i)
+
+  out = A * B * C * D * E
+  network = [A, B, C, D, E]
+
+  nodes, dict = itensorah.generate_einsum_expr([network])
+  node = go.generate_optimal_tree(nodes[1])
+  out_list = itensorah.compute_graph([node], dict)
+  out2 = out_list[1]
+
+  @test isapprox(storage(out), storage(out2))
+end
+
+@testset "test optimal contraction path" begin
+  i = Index(2, "i")
+  j = Index(3, "j")
+  k = Index(4, "k")
+  l = Index(5, "l")
+  m = Index(6, "m")
+
+  A = randomITensor(i, j)
+  B = randomITensor(j, k)
+  C = randomITensor(k, l)
+  D = randomITensor(l, m)
+  E = randomITensor(m, i)
+
+  out = A * B * C * D * E
+
+  network = itensorah.generate_optimal_tree([A, B, C, D, E])
+  out2 = contract(network)
+
+  @test isapprox(storage(out), storage(out2))
+end
+
+@testset "test gradient" begin
+  i = Index(2, "i")
+  j = Index(3, "j")
+  k = Index(4, "k")
+  l = Index(5, "l")
+  m = Index(6, "m")
+
+  A = randomITensor(i, j)
+  B = randomITensor(j, k)
+  C = randomITensor(k, l)
+  D = randomITensor(l, m)
+  E = randomITensor(m, i)
+
+  gradA_direct = B * C * D * E
+  gradB_direct = A * C * D * E
+
+  networks = itensorah.gradients([A, B, C, D, E], [A, B])
+  gradA = contract(networks[1])
+  gradB = contract(networks[2])
+
+  @test isapprox(norm(gradA_direct), norm(gradA))
+  @test isapprox(norm(gradB_direct), norm(gradB))
```

Comment:
Seems better to compare the actually tensors, not just the norm.

Reply:
Done